src/lib.rs

```diff
@@ -89,6 +97,23 @@ impl Service<Request<Body>> for MetricProvider {
     }
 }
 
+fn is_auth_token_valid(secret: &str, headers: &HeaderMap<HeaderValue>) -> bool {
+    match headers.get(AUTHORIZATION) {
+        Some(value) => value
+            .to_str()
+            .ok()
+            .map(|t| {
+                if let Some(t) = t.strip_prefix("Bearer") {
```

Comment:
This would accept `Authorization: Bearerfoo` as valid if the token was `foo`. You should strip the `Bearer ` (with the space) prefix instead.

Reply:
done